imap.go

```diff
@@ -1,102 +1,182 @@
 package maddy
 
 import (
-	"io"
-	"log"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
 	"os"
+	"strings"
+	"sync"
 
-	imapcompress "github.com/emersion/go-imap-compress"
-	imapproxy "github.com/emersion/go-imap-proxy"
 	imapbackend "github.com/emersion/go-imap/backend"
 	imapserver "github.com/emersion/go-imap/server"
-	imappgp "github.com/emersion/go-pgpmail/imap"
-	pgplocal "github.com/emersion/go-pgpmail/local"
-	"github.com/mholt/caddy/caddyfile"
+	"github.com/emersion/maddy/config"
+	"github.com/emersion/maddy/log"
+	"github.com/emersion/maddy/module"
+
+	appendlimit "github.com/emersion/go-imap-appendlimit"
+	compress "github.com/emersion/go-imap-compress"
+	move "github.com/emersion/go-imap-move"
+	"github.com/foxcpp/go-sqlmail/imap/children"
```

Comment:
We could probably move this to maddy.

I've also been considering adding an `ext` package to go-imap.

Reply:
Ah, circular dependencies…